Fix Go imports

nebolsin opened the pull request:
Those imports confuse `go mod tidy` leading to extra packages in `stellar/go` dependencies (see https://github.com/stellar/go/blob/master/go.mod#L52 vs. https://github.com/stellar/go/blob/master/go.mod#L72).

/cc @bartekn

nebolsin committed:
Fix Go imports

bartekn commented:
@leighmcculloch can you check it out?

leighmcculloch commented:
We should only do this if we're treating this fork as the canonical version of `go-xdr`, which we're sort of doing the way we're importing it. @bartekn Do you know if we still have plans to propose the changes in `nullstyle/go-xdr` back upstream to `davecgh/go-xdr`? There are about 24 commits there.

leighmcculloch commented:
I'm exploring in davecgh/go-xdr#17 if @davecgh would be open to us proposing our changes upstream. If we don't hear back in a few days (let's say Oct 25th) we'll assume that we should continue to develop go-xdr here and I'll merge this change that updates the import paths internally.

nebolsin commented:
@leighmcculloch Sounds good. I'd like to point out that the main change of this PR is switching from `github.com/nullstyle/go-xdr` to `github.com/stellar/go-xdr` in `xdr3` package tests (as per [this decision](https://github.com/nullstyle/go-xdr/pull/8#issuecomment-421164020)). I updated the import path in `xdr` and `xdr2` for consistency, but that doesn't affect `stellar/go` in any way.

leighmcculloch commented on .travis.yml:
```diff
@@ -1,19 +1,12 @@
-  - go get -v $gotools/cmd/cover
-  - go get -v $gotools/cmd/vet
-  - go get -v github.com/bradfitz/goimports
-  - go get -v github.com/golang/lint/golint
```
What's the motivation for changing these?

nebolsin replied:
- `$gotools` trick is for Go < 1.4, so we probably don't need it anymore 
- I believe `vet` and `cover` tools are pre-installed by default now
- `github.com/bradfitz/goimports` moved to `golang.org/x/tools/cmd/goimports`
- `github.com/golang/lint/golint` moved to `golang.org/x/lint/golint`

leighmcculloch commented on .travis.yml:
```diff
@@ -1,19 +1,12 @@
-  - go get -v $gotools/cmd/vet
-  - go get -v github.com/bradfitz/goimports
-  - go get -v github.com/golang/lint/golint
+  - go get -v github.com/golangci/golangci-lint/cmd/golangci-lint
```
Could we stick with using the tools independently rather than with golangci-lint?

leighmcculloch reviewed (commented):
@nebolsin I'd like to merge your change, but can we separate the changes to the CI from the changes to the import paths? I'd like to merge this one it is hyper-focused on solving the problem. If CI isn't passing with the existing build setup, can we move that to another PR that we merge first?

nebolsin commented:
@leighmcculloch I updated this PR to only include import path fixes and extracted Travis fixes to #5. Unfortunately, I cannot make `go vet` to pass without import path changes, so I'm not sure if we can fix the build before merging this PR :(

leighmcculloch commented:
Is this PR needed any longer given that #5 ended up incorporating the import path changes?

leighmcculloch commented:
@nebolsin I'm closing this because I think it was addressed by your other PR #5. Please reopen if I'm missing something.
